### PR TITLE
Create WORKING-AGREEMENTS.md

### LINKED ISSUE #2249: document working agreements

### PR DESCRIPTION
**Pull request by homeyer**
closes #2249

### CODE REVIEW

**Comment by nparsons08** [comment]
👍

File: WORKING-AGREEMENTS.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+### Don't deploy late in the day without prior agreement from the team.
+Particularly Fridays at 6pm :). At Waffle we continuously deploy our code to production; a merged PR will be seen by customers in less than 5 minutes. When you merge a PR, it's your responsibility, with help from the team, to monitor for any unforseen customer performance or usability impacts. That's hard to do when we hit merge and walk out the door.
+
+### Clear Acceptance Criteria and Sizing before a card is moved into "Current Iteration"
+As our team grows, it's important to make sure the work we do is clearly defined before we do it. The AC and the Size are determined by the team, usually during a weekly planning meeting. This working agreement helps us all stay on the same page with what we're committing to during an iteration.
+
+### Use the `expedite` label for any work that's pulled into "In Progress" and supercedes current work.
+For example, an important customer reported bug might need to be expedited. Put the `expedited` label on it, so the whole team knows that we're blocking other work to jump on it.
```
Comment (marydavis) [inline WORKING-AGREEMENTS.md]: recommend changing "an important customer" to just "a customer" - aren't all of our customers important :)

File: WORKING-AGREEMENTS.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+### Talk about any work that's moved to "In Progress" or "Current Iteration" outside of a planning meeting, either in Slack or during Standup.
```
Comment (marydavis) [inline WORKING-AGREEMENTS.md]: what do you think about not being specific to the tool but just 'team chat room'

File: WORKING-AGREEMENTS.md (hunk 1 above)
Comment (burtonjc) [inline WORKING-AGREEMENTS.md]: +1 to that

File: WORKING-AGREEMENTS.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+We plan in 2-week increments (called Iterations). It's common for interrupt work to be added to the current iteration based on customer support needs. For any interrupt work, make sure we talk about it to keep the team in sync.
+
+### PRs for all code changes
+Since Waffle is continuously deployed via [Codeship](https://codeship.com) when anything is committed to master, it's important for us to use Pull Requests to review code and make sure automated tests pass. Committing directly to master is only acceptable for emergencies, and should still be reviewed by a peer.
```
Comment (burtonjc) [inline WORKING-AGREEMENTS.md]: If we are going to be generic about Slack, we should be generic about Codeship too.

File: WORKING-AGREEMENTS.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+### All PRs should be code reviewed and functionally reviewed before merging
+Ideally PRs should also be deployed to our staging environment, if possible. 
+
+### Review PRs after standup, if not sooner
+As a distributed team, it's sometimes easy to miss reviewing a peer's PR. We don't want to stop the flow of work to have PRs reviewed immediately, but they should be looked at right after our morning Standup if they're still un-reviewed. The Waffle "Needs Review" column is the best place to see if there are open PRs that need someone's attention.
+
```
Comment (burtonjc) [inline WORKING-AGREEMENTS.md]: There isn't anything specific about waiting for a :+1: before it is ok to merge. Do we want that?

**Comment by ashumz** [comment]
My only suggestion is to add information/guidelines on how new working agreements get adopted and added or removed from this list

File: WORKING-AGREEMENTS.md (hunk 2 above)
Comment (homeyer) [inline WORKING-AGREEMENTS.md]: any reason why we shouldn't be open about the tools we use?

File: WORKING-AGREEMENTS.md (hunk 1 above)
Comment (homeyer) [inline WORKING-AGREEMENTS.md]: fixed

File: WORKING-AGREEMENTS.md (hunk 4 above)
Comment (homeyer) [inline WORKING-AGREEMENTS.md]: Added

**Comment by homeyer** [comment]
@ashumz, I added a section for how we change them.

File: WORKING-AGREEMENTS.md (hunk 2 above)
Comment (marydavis) [inline WORKING-AGREEMENTS.md]: i wouldn't mind if i loved Slack and would recommend it. I feel we are using Slack so we can make an integration - i wouldn't recommend it as it is. As for Codeship, we chose it as our tool and i would recommend it.

**Comment by burtonjc** [comment]
:+1:

**Comment by marydavis** [comment]
:+1:

**Comment by homeyer** [comment]
waiting on @kellihsf and @ashumz

**Comment by kellihsf** [comment]
:+1:

**Pull request merged**